.github/workflows/full_tests.yml

```diff
@@ -0,0 +1,123 @@
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Full test suite
+
+on:
+  pull_request_review:
+    types: [ submitted ]
+
+jobs:
+  change-info:
+    name: Check previous checks
+    runs-on: ubuntu-latest
+    if: >-
+      github.event.review.state == 'approved'
+      || github.event.review.body == 'recheck-snap'
+      || github.event.review.body == 'recheck-full'
+    outputs:
+      code-changed: ${{ steps.code-changed.outputs.changed }}
+      skip: ${{ steps.skip-tests.outputs.skip }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
+      - name: Get latest workflow run status
+        uses: actions/github-script@v4
+        id: latest-workflow
+        with:
+          script: |
+            const runs = await github.actions.listWorkflowRuns({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              workflow_id: 'full_tests.yml',
+              per_page: 2
+            })
+            return runs.data.workflow_runs[1]
+      - name: Check to see if there is any new confirmation
+        id: new-commit
+        env:
+          LATEST_HEAD_SHA: ${{ fromJson(steps.latest-workflow.outputs.result).head_sha }}
+        run: |
+          result=0
+          git log | head -n 2 | grep -qE $LATEST_HEAD_SHA || result=1
+          echo "::set-output name=new::$result"
+      - name: Track changed files in the PR
+        id: code-changed
+        run: |
+          git remote add upstream https://github.com/canonical/juju-verify
+          git fetch upstream master
+          failed=1
+          git diff --name-only upstream/master | grep -qE '^(./juju_verify|./tests)' || failed=0
```

Comment:
Could `fetch-depth: 2` config option for the checkout action affect this diff?

Reply:
Additionally, unit and functional tests should not run if `(./juju-verify|./tests)` folders are not edited. However, if `snap/snapcraft.yaml` changes, snaps should be tested for getting built.

It seems the workflow is correct because the `build-snaps` job does not have any additional condition (it will always run). So all good here, thanks!

Reply:
No, I don't think so. "git diff" compares the last commit and the master branch.

Only `func-tests` are using output `code-changed` to skip tests if no code was changed.
The `build-snap` is run whenever the` lint-unittests-docs-build` is ->
 - New Approval review and there is new commit or last check was not "success"
 - New review w/ `recheck-snap` comment
 - New review w/ `recheck-full` comment